toolkit/tools/internal/retry/retry.go

```diff
@@ -9,10 +9,10 @@ import (
 )
 
 const (
-	// With 5 attempts (4 retries) and a backoff factor of 2 seconds the total time spent retrying will be approximately:
-	// 1 + 4 + 8 + 16 = 31 seconds.
+	// With 8 attempts (7 retries) and a backoff factor of 2 seconds the total time spent retrying will be approximately:
```

Comment:
There are a few spots where this will cause us problems were we don't check for 404. The short retry wasn't a huge issue but >2min doesn't feel good. Let me have a quick poke at this.

Reply:
Sure, let me know what you'll find out and the next steps on this please.

Reply:
This got a bit out of hand but... https://github.com/microsoft/azurelinux/pull/9251